roles/openshift_health_checker/openshift_checks/docker_storage_check.py

```diff
@@ -0,0 +1,137 @@
+    tags = ["preflight"]
+
+    storage_drivers = ["devicemapper", "overlay2"]
+
+    max_thinpool_data_usage_percent = 90.0
+    max_thinpool_metadata_usage_percent = 90.0
+
+    def run(self, tmp, task_vars):
+        self.max_thinpool_data_usage_percent = get_var(task_vars, "max_thinpool_data_usage_percent", default=self.max_thinpool_data_usage_percent)
+        self.max_thinpool_metadata_usage_percent = get_var(task_vars, "max_thinpool_metadata_usage_percent", default=self.max_thinpool_metadata_usage_percent)
+
+        info = self.execute_module("docker_info", {}, task_vars).get("info", {})
+
+        if not self.is_supported_storage_driver(info):
+            msg = "Unsupported Docker storage driver detected. Supported storage drivers: {drivers}"
+            return {"failed": True, "msg": msg.format(drivers=self.storage_drivers)}
+
+        if self.is_using_loopback_device(info):
+            msg = "Use of loopback devices is discouraged. Try running Docker with `--storage-opt dm.thinpooldev`"
+            return {"failed": True, "msg": msg}
+
+        failed, msg = self.check_thinpool_usage(task_vars)
+        if failed:
+            return {"failed": True, "msg": msg}
+
+        return {"changed": False}
+
+    def is_supported_storage_driver(self, docker_info):
+        return docker_info.get("Driver", "") in self.storage_drivers
+
+    @staticmethod
+    def is_using_loopback_device(docker_info):
+        # Loopback device usage is only an issue if using devicemapper.
+        # Skip this check if using any other storage driver.
+        if docker_info.get("Driver", "") != "devicemapper":
+            return False
+
+        for status in docker_info.get("DriverStatus", []):
+            if status[0] == "Data loop file":
+                return bool(status[1])
+
+        return False
+
+    def check_thinpool_usage(self, task_vars):
+        no_data_msg = "no thinpool usage data returned by the host."
+
+        failed, data, msg = self.get_lvs_data(task_vars)
+        if failed:
+            if not msg:
+                msg = no_data_msg
+            return failed, msg
+
+        failed, data_percent = self.get_thinpool_data_usage(data)
+        if failed:
+            return failed, no_data_msg
+
+        failed, metadata_percent = self.get_thinpool_metadata_usage(data)
+        if failed:
+            return failed, no_data_msg
+
+        if data_percent > self.max_thinpool_data_usage_percent:
+            msg = "thinpool data usage above maximum threshold of {threshold}%%"
+            return True, msg.format(threshold=data_percent)
+
+        if metadata_percent > self.max_thinpool_metadata_usage_percent:
+            msg = "thinpool metadata usage above maximum threshold of {threshold}%%"
+            return True, msg.format(threshold=metadata_percent)
+
+        return False, ""
+
+    def get_lvs_data(self, task_vars):
+        lvs_cmd = "/sbin/lvs --select vg_name=docker --select lv_name=docker-pool --report-format json"
```

Comment:
I attempted to look for a core module to use in place of this command, however the closest one I found was http://docs.ansible.com/ansible/lvol_module.html, which does not seem to return the information needed for this check